### PR TITLE
Report failed (non-staking) transactions

### PR DESCRIPTION
**Pull request by Daniel-VDM**
## Issue

Similar to #2157 #2151, this PR will report failed (non-staking) transactions on all nodes instead of just on the leader. 

### Unit Test Coverage

Before:

```
PASS
coverage: 13.6% of statements
ok  	github.com/harmony-one/harmony/node	2.391s
```

```
PASS
coverage: 19.5% of statements
ok  	github.com/harmony-one/harmony/core	6.853s
```

After:

```
PASS
coverage: 14.0% of statements
ok  	github.com/harmony-one/harmony/node	2.067s
```

```
PASS
coverage: 20.4% of statements
ok  	github.com/harmony-one/harmony/core	6.656s
```

## Operational Checklist

1. **Does this PR introduce backward-incompatible changes to the on-disk data structure and/or the over-the-wire protocol?**. (If no, skip to question 8.)

    **NO**

8. **Does this PR introduce backward-incompatible changes *NOT* related to on-disk data structure and/or over-the-wire protocol?** (If no, continue to question 11.)

    **NO**

11. **Does this PR introduce significant changes to the operational requirements of the node software, such as >20% increase in CPU, memory, and/or disk usage?**

    **NO**

### CODE REVIEW

File: core/tx_pool.go [hunk 1]
```diff
@@ -18,14 +18,15 @@ package core
...
-	"errors"
 	"fmt"
 	"math"
 	"math/big"
 	"sort"
 	"sync"
 	"time"
 
+	"github.com/pkg/errors"
+
```
Comment (fxfactorial) [inline core/tx_pool.go]: prefer no whitespace between packages

File: core/tx_pool.go [hunk 2]
```diff
@@ -884,13 +899,17 @@ func (pool *TxPool) addTxsLocked(txs []*types.Transaction, local bool) []error {
 	// Add the batch of transaction, tracking the accepted ones
 	dirty := make(map[common.Address]struct{})
 	errs := make([]error, len(txs))
+	var erroredTxns []types.RPCTransactionError
```
Comment (fxfactorial) [inline core/tx_pool.go]: prefer short declaration, errorTxns := []foo{}

File: core/tx_pool.go (hunk 1 above)
Comment (Daniel-VDM) [reply core/tx_pool.go]: fixed in latest

File: core/tx_pool.go (hunk 2 above)
Comment (Daniel-VDM) [reply core/tx_pool.go]: fix in latest

**Pull request merged**